eng/native/init-compiler-and-cmake.cmd

```diff
@@ -0,0 +1,89 @@
+@if not defined _echo @echo off
```

Comment:
this file was extracted from https://github.com/dotnet/runtime/blob/master/src/libraries/Native/build-native.cmd and/or https://github.com/dotnet/runtime/blob/master/src/installer/corehost/build.cmd, we can unify these later on.